examples/petsc/ex28p.cpp

```diff
@@ -0,0 +1,401 @@
+   std::cout << "Solving...\n";
+
+   trueX = 0.0;
+
+   SlepcEigenSolver *solver = new SlepcEigenSolver(MPI_COMM_WORLD);
+   solver->SetOperators(*LHSOp,*RHSOp);
+   solver->SetNumModes(nev);
+   solver->SetWhichEigenpairs(SlepcEigenSolver::TARGET_MAGNITUDE);
+   // The target is set with a small offset to prevent zero pivots in this example
+   solver->SetTarget(pow(k0*2.0,2)-1e-2);
+   solver->Solve();
+   double re;
+   solver->GetEigenvalue(0,re);
+   Vector dummy2(block_trueOffsets[2]);
+   solver->GetEigenvector(0,trueX);
+   std::cout << "Effective index: " << sqrt(re)/k0 << "\n";
+
+   // 12. Extract the parallel grid function corresponding to the finite element
+   //     approximation X. This is the local solution on each processor.
+   ParGridFunction *et(new ParGridFunction);
+   ParGridFunction *ez(new ParGridFunction);
+   et->MakeRef(N_space, x.GetBlock(0), 0);
+   ez->MakeRef(L_space, x.GetBlock(1), 0);
+   et->Distribute(&(trueX.GetBlock(0)));
+   ez->Distribute(&(trueX.GetBlock(1)));
+
+   // 13. Save the refined mesh and the solution in parallel. This output can be
+   //     viewed later using GLVis: "glvis -np <np> -m mesh -g sol_*".
+   {
+      ostringstream mesh_name, u_name, p_name;
+      mesh_name << "mesh." << setfill('0') << setw(6) << myid;
+      u_name << "sol_u." << setfill('0') << setw(6) << myid;
+      p_name << "sol_p." << setfill('0') << setw(6) << myid;
+
+      ofstream mesh_ofs(mesh_name.str().c_str());
+      mesh_ofs.precision(8);
+      pmesh->Print(mesh_ofs);
+
+      ofstream et_ofs(u_name.str().c_str());
+      et_ofs.precision(8);
+      et->Save(et_ofs);
+
+      ofstream ez_ofs(p_name.str().c_str());
+      ez_ofs.precision(8);
+      ez->Save(ez_ofs);
+   }
+
+   // 14. Save data in the VisIt format
+   VisItDataCollection visit_dc("Example28-Parallel", pmesh);
+   visit_dc.RegisterField("Exy", et);
+   visit_dc.RegisterField("Ez", ez);
+   visit_dc.SetFormat(!par_format ?
+                      DataCollection::SERIAL_FORMAT :
+                      DataCollection::PARALLEL_FORMAT);
+   visit_dc.Save();
+
+   // 15. Send the solution by socket to a GLVis server.
+   if (visualization)
+   {
+      char vishost[] = "localhost";
+      int  visport   = 19916;
+      socketstream u_sock(vishost, visport);
+      u_sock << "parallel " << num_procs << " " << myid << "\n";
+      u_sock.precision(8);
+      u_sock << "solution\n" << *pmesh << *et << "window_title 'Transverse E field'"
+             << endl;
+      u_sock << "keys Rjl!\n";
+      // Make sure all ranks have sent their 'et' solution before initiating
+      // another set of GLVis connections (one from each rank):
+      MPI_Barrier(pmesh->GetComm());
+      socketstream p_sock(vishost, visport);
+      p_sock << "parallel " << num_procs << " " << myid << "\n";
+      p_sock.precision(8);
+      p_sock << "solution\n" << *pmesh << *ez << "window_title 'Longitudinal E field'"
+             << endl;
+      p_sock << "keys Rjl!\n";
```

Comment:
I'm not familiar with the "!" GLVis key but it makes my "Longitudinal E field" plots solid black.  In order to see anything I need to hit 'p' to switch the palette.  In the other plot I don't see an effect.  This may indicate a problem with GLVis on my laptop but you may want to consider removing the '!' unless you really want it.

Reply:
The "!" key switches between discret, smooth and no texture, see https://github.com/GLVis/glvis/blob/master/README#L159-L164.

Its effect is most pronounced in pallets such as 5-color (first choice with `F6`).

If you are getting solid black, that's a bug. Can you report a reproducer on the GLVis issue tracker?

Reply:
@tzanio did find a bug in GLVis which lead to this problem.  No changes are necessary in this PR.